lib/graphiti/util/persistence.rb

```diff
@@ -43,18 +44,16 @@ def initialize(resource, meta, attributes, relationships, caller_model, foreign_
   #
   # @return a model instance
   def run
-    parents = process_belongs_to(@relationships)
-    update_foreign_key_for_parents(parents)
+    attributes = @adapter.persistence_attributes(self, @attributes)
```

Comment:
adding a hook to get a chance of modifying attributes via passing them in adapter here.
In case of activegraph we are merging association attributes with model attributes here so that associations get assigned directly while object is being persisted. 

where as in `ActiveRecord` they are looping over through associations and assigning them with parent object later in this method using `process_belongs_to` and `process_has_many`